modules/dcache-chimera/src/main/java/org/dcache/chimera/namespace/ChimeraCleaner.java

```diff
@@ -454,20 +457,34 @@ public void messageArrived(PoolManagerPoolUpMessage poolUpMessage)
         }
     }
 
+    /**
+     * runNotification
+     * collect messages from database and send Delete Notifications
+     */
     private void runNotification()
             throws InterruptedException
     {
         final String QUERY =
                 "SELECT ipnfsid FROM t_locationinfo_trash t1 " +
-                "WHERE itype=2 AND NOT EXISTS (SELECT 1 FROM t_locationinfo_trash t2 WHERE t2.ipnfsid=t1.ipnfsid AND t2.itype <> 2)";
-        for (String id : _db.queryForList(QUERY, String.class)) {
-            try {
-                sendDeleteNotifications(new PnfsId(id)).get();
-                _db.update("DELETE FROM t_locationinfo_trash WHERE ipnfsid=? AND itype=2", id);
-            } catch (ExecutionException e) {
-                _log.warn(e.getCause().getMessage());
+                "WHERE itype=2 AND NOT EXISTS (SELECT 1 FROM t_locationinfo_trash t2 WHERE t2.ipnfsid=t1.ipnfsid AND t2.itype <> 2) LIMIT=?";
+
+        List<String> queryList = new ArrayList<String>();
```

Comment:
The patch doesn't need to create this object -- Java should be smart enough to realise that the first line in the do-loop will always assign `queryList` a value ahead of the `while` clause.